File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include <iostream>
#include <queue>
#include <vector>
#include <algorithm>

using namespace std;

int N, M, V;
int c1[1001], c2[1001];
vector<int> v[1001];

void bfs(int start) {
	queue<int> q;
	q.push(start);
	c1[start] = true;
	while (!q.empty()) {
		int x = q.front();
		q.pop();
		printf("%d ", x);
		for (int i = 0; i < v[x].size(); i++) {
			int y = v[x][i];
			if (!c1[y]) {
				q.push(y);
				c1[y] = true;
			}
		}
	}
}

void dfs(int x) {
	if (c2[x]) return;
	c2[x] = true;
	printf("%d ", x);
	for (int i = 0; i < v[x].size(); i++) {
		int y = v[x][i];
		dfs(y);
	}
}
int main() {
	scanf("%d %d %d", &N, &M, &V);
	for (int i = 1; i <= M; i++) {
		int p1, p2;
		scanf("%d %d", &p1, &p2);
		v[p1].push_back(p2);
		v[p2].push_back(p1);
	}
	for (int i = 1; i <= N; i++)
		sort(v[i].begin(), v[i].end());
	dfs(V);
	printf("\n");
	bfs(V);
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.util.*;
import java.io.*;

public class Main {
    static List<Integer>[] graph;
    static int n, m, v;
    static boolean visited[];
    
    public static String bfs(int start){
        Deque<Integer> dq = new LinkedList<>();
        dq.addFirst(start);
        visited[start] = true;

        StringBuffer sb = new StringBuffer();

        while(!dq.isEmpty()){
            int node = dq.pollLast();
            sb.append(node).append(" ");

            for(int next_node : graph[node]){
                if(!visited[next_node]){
                    visited[next_node] = true;
                    dq.addFirst(next_node);
                }
            }
        }

        return sb.toString();
    }

    public static String dfs(int node){
        StringBuffer sb = new StringBuffer();
        visited[node] = true;
        sb.append(node).append(" ");
        for(int next_node : graph[node]){
            if (!visited[next_node]){
                sb.append(dfs(next_node));
            }
        }
        return sb.toString();
    }
    
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());
        v = Integer.parseInt(st.nextToken());

        graph = new ArrayList[n+1];
        for(int i=0; i<n+1; i++){
            graph[i] = new ArrayList<Integer>();
        }

        for(int i=0; i<m; i++){
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());

            graph[a].add(b);
            graph[b].add(a);
        }

        for(int i=0; i<n+1; i++){
            Collections.sort(graph[i]);
        }

        visited = new boolean[n+1];
        System.out.println(dfs(v));
        visited = new boolean[n+1];
        System.out.println(bfs(v));
    }
}